src/jit/lower.cpp

```diff
@@ -1084,25 +1074,22 @@ void Lowering::LowerArg(GenTreeCall* call, GenTreePtr* ppArg)
             NYI("Lowering of long register argument");
         }
 
-        // For longs, we will create two PUTARG_STKs below the GT_LONG. The hi argument needs to
-        // be pushed first, so the hi PUTARG_STK will precede the lo PUTARG_STK in execution order.
+        // For longs, we will replace the GT_LONG with a GT_FIELD_LIST, and put that under a PUTARG_STK.
+        // Although the hi argument needs to be pushed first, that will be handled by the general case,
+        // in which the fields will be reversed.
         noway_assert(arg->OperGet() == GT_LONG);
```

Comment:
This should probably be a plain `assert`: barring an inlining change, we're unlikely to be able to avoid passing a long-typed argument simply by disabling the optimizer.